Source/EventFlow/Subscribers/IDispatchToSubscriberLog.cs

```diff
@@ -0,0 +1,50 @@
+﻿// The MIT License (MIT)
+// 
+// Copyright (c) 2015-2020 Rasmus Mikkelsen
+// Copyright (c) 2015-2020 eBay Software Foundation
+// https://github.com/eventflow/EventFlow
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EventFlow.Aggregates;
+
+namespace EventFlow.Subscribers
+{
+    public interface IDispatchToSubscriberLog
+    {
+        Task HandleEventBeginAsync(
+            ISubscribeTo subscriberTo,
+            IDomainEvent domainEvent,
```

Comment:
I'm affraid that we should store fact information on every event. Can we receive here collection of `IDomainEvent`? And in `HandleEventFailed` method - collection of failed events and in `HandleEventDone` - collection of done events even exception was thrown?

Reply:
Domain events are published to subscribers in order. Thus if an aggregate emits both `EventA` and `EventB`, then `EventA` is sent to all subscribers before `EventB` is sent. We would need to change the order if we wanted to support this.

Maybe something can be done.